packages/wallet/src/services/TransactionsTracker.ts

```diff
@@ -160,6 +160,7 @@ export const createAddressTransactionsProvider = ({
                 );
                 const duplicateTransactions =
                   lastStoredTransaction && intersectionBy(localTransactions, newTransactions, (tx) => tx.id);
+
                 if (typeof duplicateTransactions !== 'undefined' && duplicateTransactions.length === 0) {
                   const rollbackTransactions = localTransactions.filter(
```

Comment:
Following the logic from prev comment, we rollback one transaction at a time, so:
```diff
- const rollbackTransactions = localTransactions.filter(
+ const rollbackTransactions = [lastStoredTransaction]
```